### DISCUSSION TITLE
Enable the Create CQRS CRUD Operations context menu in folders

### DISCUSSION THREAD

**Discussion by stephanjohnson** [IntentArchitect/Support#29] [General]
Currently the `Create CQRS CRUD Operations` context menu is only available at the root of the services package. It would be useful if this option can be available on folders in the service designer. If we want to order our services using folders, we have to run this command at the root of the service package, and manually move the resulting Commands, Queries and DTO's into the desired folder. 

This has the side-effect of routes not being aligned with the folder structure and having to manually update each route to match the folder structure.

**Comment by joelsteventurner** [comment]
Hi @stephanjohnson

Another great suggestion, I have logged it as an enhancement on our back log.

**Comment by joelsteventurner** [comment]
Hi @stephanjohnson 

Just let you know this feature is now available in pre-release.
Intent.Application.MediatR.CRUD 5.1.4-pre.0

Try it out and let me know what you think.

**Comment by stephanjohnson** [comment]
Hi @joelsteventurner 

I have tested the `5.1.4-pre.0` release, and I can create CRUD operations from a folder.

> This has the side-effect of routes not being aligned with the folder structure and having to manually update each route to match the folder structure.

This is not happening yet, and hoping for #28 to also become a feature. Looks great so far.

**Comment by joelsteventurner** [comment]
Hi @stephanjohnson 

I forgot to mention, the folder route alignment is also available in pre-release.
Intent.Metadata.WebApi 4.2.4-pre.0

**Comment by stephanjohnson** [comment]
Hi @joelsteventurner 

I've updated the the `Intent.Metadata.WebApi 4.2.4-pre.0` and it works

![image](https://github.com/IntentArchitect/Support/assets/6469385/88704791-eb4f-447d-95ab-d91376ff75b8)

Also tested out how you'd generate the slug for `folder 1` vs `folder-2` and it is perfect
